From 90b4cb933a7764298d42ae81722029b3bc2fa74f Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Tue, 09 Jun 2020 08:37:20 +0300
Subject: [PATCH] [SW-11875] goya2: ppboot: Switch to new flow

This patch adds changes to switch download and verification of preboot
from arcboot to ppboot.

Change-Id: I6a547d23fac594045f34ab09a953d935844839f2
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 98ae2cd..90cd82f 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -205,11 +205,6 @@
 		;
 }
 
-int hl_pcie_init_only(void)
-{
-	return 0;
-}
-
 void hl_plat_init(void) {}
 
 
diff --git a/goya/common.c b/goya/common.c
index 13b862f..dc783f4 100644
--- a/goya/common.c
+++ b/goya/common.c
@@ -161,11 +161,6 @@
 
 void preboot_sync_btl(void) {}
 
-int hl_pcie_init_only(void)
-{
-	return 0;
-}
-
 void hl_plat_init(void) {}
 
 void image_start(u64 loadaddr)
diff --git a/goya2/common.c b/goya2/common.c
index e216bb7..07cc248 100644
--- a/goya2/common.c
+++ b/goya2/common.c
@@ -201,11 +201,6 @@
 		;
 }
 
-int hl_pcie_init_only(void)
-{
-	return 0;
-}
-
 void hl_plat_init(void) {}
 
 void ppboot_log_waiting_for_pcie(void)
diff --git a/goya2_arc/common.c b/goya2_arc/common.c
index 83ebb69..68dc2c9 100644
--- a/goya2_arc/common.c
+++ b/goya2_arc/common.c
@@ -226,12 +226,6 @@
 
 void preboot_sync_btl(void) {}
 
-
-int hl_pcie_init_only(void)
-{
-	return 1;
-}
-
 void init_prstn(void)
 {
 	/* Configure PRSTN to reset entire SoC */
diff --git a/goya2_arc_gnu/common.c b/goya2_arc_gnu/common.c
index dfbe0df..6db9cfe 100644
--- a/goya2_arc_gnu/common.c
+++ b/goya2_arc_gnu/common.c
@@ -84,12 +84,6 @@
 
 void preboot_sync_btl(void) {}
 
-
-int hl_pcie_init_only(void)
-{
-	return 1;
-}
-
 void spi_ssi_cs0_get(unsigned int *mask, unsigned int *data_offset,
 		     unsigned int *out_en_offset)
 {
diff --git a/include/preboot.h b/include/preboot.h
index 99d00b1..79c621a 100644
--- a/include/preboot.h
+++ b/include/preboot.h
@@ -89,7 +89,6 @@
 u32 hl_skip_pcie_init(void);
 void hl_uart_init(void);
 u32 hl_spi_baudr_get(void);
-int hl_pcie_init_only(void);
 void hl_plat_init(void);
 void hl_ts_init(void);
 
diff --git a/ppboot/ppboot.c b/ppboot/ppboot.c
index 6f84dab..15a1c9b 100644
--- a/ppboot/ppboot.c
+++ b/ppboot/ppboot.c
@@ -167,10 +167,6 @@
 	ppboot_version();
 	rst_src_dump();
 
-	/* check if we are done here */
-	if (hl_pcie_init_only())
-		return;
-
 	preboot_delay();
 	loadaddr = copy_preboot();
 
